tests/unit/test_validate_attribute.py

```diff
@@ -0,0 +1,1115 @@
+        self, va_obj: ValidateAttribute, cross_val_df1: DataFrame
+    ):
+        """Tests for cross manifest validation for matchAtLeastOne set rules"""
+        val_rule = "matchAtLeastOne Patient.PatientID set error"
+
+        with patch.object(
+            schematic.models.validate_attribute.ValidateAttribute,
+            "_get_target_manifest_dataframes",
+            return_value={"syn1": cross_val_df1},
+        ):
+            assert va_obj.cross_validation(val_rule, Series(["A", "B", "C"])) == (
+                [],
+                [],
+            )
+            assert va_obj.cross_validation(val_rule, Series(["A", "B", "C", "C"])) == (
+                [],
+                [],
+            )
+            assert va_obj.cross_validation(
+                val_rule, Series(["A", "B", "C", "A", "B", "C"])
+            ) == (
+                [],
+                [],
+            )
+            assert va_obj.cross_validation(
+                val_rule, Series(["A", "B"], index=[0, 1], name="PatientID")
+            ) == (
+                [],
+                [],
+            )
+            assert va_obj.cross_validation(
+                val_rule, Series([], index=[], name="PatientID")
+            ) == (
+                [],
+                [],
+            )
+
+    def test_cross_validation_match_atleast_one_set_rules_errors(
+        self, va_obj: ValidateAttribute, cross_val_df1: DataFrame
+    ):
+        """Tests for cross manifest validation for matchAtLeastOne set rules"""
+        val_rule = "matchAtLeastOne Patient.PatientID set error"
+
+        with patch.object(
+            schematic.models.validate_attribute.ValidateAttribute,
+            "_get_target_manifest_dataframes",
+            return_value={"syn1": cross_val_df1},
+        ):
+            errors, warnings = va_obj.cross_validation(
+                val_rule,
+                Series(["A", "B", "C", "D"], index=[0, 1, 2, 3], name="PatientID"),
+            )
+            assert len(warnings) == 0
+            assert len(errors) == 1
+
+            errors, warnings = va_obj.cross_validation(
+                val_rule, Series([""], index=[0], name="PatientID")
+            )
+            assert len(warnings) == 0
+            assert len(errors) == 1
+
+    def test_cross_validation_match_atleast_one_set_rules_warnings(
+        self, va_obj: ValidateAttribute, cross_val_df1: DataFrame
+    ):
+        """Tests for cross manifest validation for matchAtLeastOne set rules"""
+        val_rule = "matchAtLeastOne Patient.PatientID set warning"
+
+        with patch.object(
+            schematic.models.validate_attribute.ValidateAttribute,
+            "_get_target_manifest_dataframes",
+            return_value={"syn1": cross_val_df1},
+        ):
+            errors, warnings = va_obj.cross_validation(
+                val_rule,
+                Series(["A", "B", "C", "D"], index=[0, 1, 2, 3], name="PatientID"),
```

Comment:
Define comfortable! :) I'm more comfortable with one than with zero which is where we were starting at.

Reply:
I feel like when scope is set to level and warning, they should have the same number of tests but just raised in different way?

Reply:
Oh right, this is just to make sure the warnings work. (They function exactly the same as errors, but if the paramater is set to `warning` they come out in the second list rather than the first, so this doesn't need any other tests.